### ISSUE TITLE
DOI status pending while DOI is active (pt 3?)

### ISSUE THREAD

**Issue by ClaraCogswell** [hydroshare/hydroshare#4878]
**Description of the bug**
DOI status says pending on resource landing page while DOI is active: https://doi.org/10.4211/hs.632e550deaea46b080bdae986fd19156

Steps to reproduce the bug:
1. Go to https://doi.org/10.4211/hs.632e550deaea46b080bdae986fd19156

**Expected behavior**
DOI should not stay status pending

**Additional information**
I couldn't find the previous ticket for this issue, apologies. If anyone can please link it

**Comment by devincowan** [comment]
https://github.com/hydroshare/hydroshare/issues/4605

**Comment by devincowan** [comment]
```
# grep 632e550deaea46b080bdae986fd19156 django.log
[05/Dec/2022 15:34:18] WARNING [django.request:230] Forbidden: /hsapi/resource/632e550deaea46b080bdae986fd19156/file_list/
[05/Dec/2022 15:34:18] WARNING [django.request:230] Forbidden: /hsapi/resource/632e550deaea46b080bdae986fd19156/file_list/
[05/Dec/2022 15:51:58] WARNING [django.request:230] Forbidden: /hsapi/resource/632e550deaea46b080bdae986fd19156/file_list/
[05/Dec/2022 15:53:32] ERROR [django.request:230] Forbidden (Permission denied): /hsapi/_internal/632e550deaea46b080bdae986fd19156/publish/
```

**Comment by devincowan** [comment]
perhaps due to:
https://github.com/celery/celery/issues/3589#issuecomment-290979095

Testing a fix

**Comment by devincowan** [comment]
confirmed fixed by [2.0.2 hotfix](https://github.com/hydroshare/hydroshare/releases/tag/2.0.2)

**Comment by martinseul** [comment]
not sure if this is related but there is another issue with a resource and a DOI please see https://www.hydroshare.org/resource/ebd3c73f946344d8adf0431f20576d1c/ cross ref complaints that there is not have a published date in it's metadata

**Comment by devincowan** [comment]
strange, this resource shows up in Discover
<img width="1465" alt="image" src="https://user-images.githubusercontent.com/17934193/206767047-51f807cf-84db-4a1f-8474-3aa20c7a2c1a.png">
and on Beta:
https://beta.hydroshare.org/resource/ebd3c73f946344d8adf0431f20576d1c/
But for me, 
https://www.hydroshare.org/resource/ebd3c73f946344d8adf0431f20576d1c/
just gives an error page. Looking into it
```
$ grep ebd3c73f946344d8adf0431f20576d1c django.log
[28/Nov/2022 17:28:24] ERROR [django.request:230] Forbidden (Permission denied): /hsapi/_internal/ebd3c73f946344d8adf0431f20576d1c/publish/
[28/Nov/2022 17:28:24] ERROR [django.request:230] Internal Server Error: /hsapi/_internal/ebd3c73f946344d8adf0431f20576d1c/publish/
[28/Nov/2022 17:28:28] ERROR [django.request:230] Forbidden (Permission denied): /hsapi/_internal/ebd3c73f946344d8adf0431f20576d1c/publish/

$ grep ebd3c73f946344d8adf0431f20576d1c hydroshare.log
[28/Nov/2022 17:24:58] INFO [root:265] Generic aggregation was created for file:ebd3c73f946344d8adf0431f20576d1c/data/contents/Supporting_Information_SI_-_2022WR032866.pdf.
```

**Comment by devincowan** [comment]
related to #4795

**Comment by devincowan** [comment]
https://api.crossref.org/v1/works/10.4211/hs.ebd3c73f946344d8adf0431f20576d1c
everything looks okay

**Comment by devincowan** [comment]
```
>>> for ra in ResourceAccess.objects.filter(published=True).all():
...     if not ra.resource.metadata.dates.filter(type='published').all().first():
...             print(ra.resource.short_id)
...

ebd3c73f946344d8adf0431f20576d1c

**Comment by devincowan** [comment]
```
>>> from hs_access_control.models import ResourceAccess
>>> for ra in ResourceAccess.objects.filter(published=True).all():
...     if not ra.resource.metadata.dates.filter(type='published').all().first():
...             ra.resource.metadata.create_element('date', type='published', start_date=ra.resource.updated)
```
The problem with https://www.hydroshare.org/resource/ebd3c73f946344d8adf0431f20576d1c/ seems more related to https://github.com/hydroshare/hydroshare/issues/4795 so I'm closing this issue and reopening #4795
